### PR TITLE
BUG: .count() raises if use_inf_as_na is enabled

### LINKED ISSUE #29478: BUG: Series.count() raises exception after upgrading from v0.24.1 to v0.25.3  if use_inf_as_na is enabled for a DateTime series.
#### Code Sample, a copy-pastable example if possible

```python
# Your code here
import pandas as pd
from datetime import datetime

if __name__ == '__main__':
    s = pd.Series([datetime.now()])
    with pd.option_context('use_inf_as_na', True):
        s.count()
```
#### Problem description

Previously with pandas v0.24.1, the above code works well,
however, after I upgrade the version to v0.25.3,
the above code raises an exception:
`AttributeError: 'DatetimeArray' object has no attribute '_constructor'`

It seems that something goes wrong when passing a `DatetimeArray` to `_isna_old`.
I think it is a bug introduced in some recent update.

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.6.8.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
machine          : AMD64
processor        : Intel64 Family 6 Model 85 Stepping 4, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.None

pandas           : 0.25.3
numpy            : 1.17.3
pytz             : 2019.3
dateutil         : 2.8.0
pip              : 18.1
setuptools       : 40.6.2
Cython           : None
pytest           : 4.3.0
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.10.3
IPython          : 7.9.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 0.12.1
pytables         : None
s3fs             : None
scipy            : 1.2.1
sqlalchemy       : None
tables           : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None

</details>

### PR DESCRIPTION
**Pull request by thy09**
- [x] closes #29478
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/series/test_analytics.py [hunk 1]
```diff
@@ -554,6 +554,10 @@ def test_count(self, datetime_series):
         ts.iloc[[0, 3, 5]] = np.nan
         tm.assert_series_equal(ts.count(level=1), right - 1)
 
+        # GH29478
+        with pd.option_context("use_inf_as_na", True):
+            pd.Series([pd.Timestamp('1990/1/1')]).count()
```
Comment (WillAyd) [inline pandas/tests/series/test_analytics.py]: Can you test the result of this expression?

**Review by WillAyd — CHANGES_REQUESTED** [review]
Nice PR thanks for the submission. Can you add a whatsnew note for 1.0.0 as well?

File: pandas/tests/series/test_analytics.py (hunk 1 above)
Comment (thy09) [reply pandas/tests/series/test_analytics.py]: Updated, thanks a lot!

**Review by jorisvandenbossche — APPROVED** [review]

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @thy09

**Comment by thy09** [comment]
It's my pleasure!

When will the next version release?
This issue caused some problems in our project now.

**Comment by jorisvandenbossche** [comment]
>  When will the next version release?

Pandas 1.0 is planned for beginning next year

**Comment by thy09** [comment]
Thanks a lot!

**Pull request merged**